### ISSUE TITLE
Duplicate individuals in new optimisation script

### ISSUE THREAD

**Issue by MatNif** [architecture-building-systems/CityEnergyAnalyst#3339]
**Describe the bug**
The new optimisation script creates duplicates of selected system configurations (i.e. identical connectivity_vector and capacity_indicator_vector) across the iterations of the optimisation algorithm.

**To Reproduce**
Steps to reproduce the behaviour:
1. Run the new optimisation script on your selected scenario (activate debugging and make sure the number of generations and population size are each set to at least 10)
2. Open the 'fitness_tracker.csv' and 'supply_system_tracker.csv' files and look for duplicate solutions (more commonly found in the later generations of the genetic algorithm)

**Expected Behaviour**
Each system configuration should only be evaluated once and appear only once in the array of selected systems.

**Screenshots**
Example of duplicate system configurations

<img width="604" alt="Screenshot 2023-07-06 133927" src="https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/93093343/e28c2155-5215-4544-aaf1-c04021a16d54">

<img width="605" alt="Screenshot 2023-07-06 134129" src="https://github.com/architecture-building-systems/CityEnergyAnalyst/assets/93093343/63455c99-6633-4fc2-b045-15dd1c449a35">

**Hardware (please complete the following information):**
 - Windows 11
 - CEA Developers Version - main branch
